### PR TITLE
Handle `set_weights()` for dtype string

### PR DESCRIPTION
**Pull request by MansBermell**
For further details see issue: https://github.com/tensorflow/tensorflow/issues/46553

### CODE REVIEW

**Comment by MansBermellLeo** [comment]
Perhaps a more elegant implementation would be preferable.

**Review by bhack — APPROVED** [review]
I approve this just to Run CI tests as I don't have label control

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/keras/backend.py [hunk 1]
```diff
@@ -1524,6 +1524,19 @@ def dtype(x):
...
 
+@doc_controls.do_not_generate_docs
+def dtype_numpy(x):
+  """Returns the numpy dtype of a Keras tensor or variable.
+
+  Args:
+      x: Tensor or variable.
+
+  Returns:
+      numpy.dtype, dtype of `x.
```
Comment (MansBermellLeo) [inline tensorflow/python/keras/backend.py:1535]: Need to add the "`". Can add when all checks are green

**Review by mihaimaruseac — APPROVED** [review]

**Comment by MansBermellLeo** [comment]
@mihaimaruseac Should I change anything due to the failed check: `import/copybara`?

**Comment by mihaimaruseac** [comment]
@MansBermellLeo no, copybara is the service that imports the PR and then merges once the internal change is approved and lands.

File: tensorflow/python/keras/backend.py [hunk 2]
```diff
@@ -1524,6 +1524,19 @@ def dtype(x):
...
+def dtype_numpy(x):
+  """Returns the numpy dtype of a Keras tensor or variable.
+
+  Args:
+      x: Tensor or variable.
+
+  Returns:
+      numpy.dtype, dtype of `x`.
+  """
+  return dtypes_module.as_dtype(x.dtype).as_numpy_dtype
```
Comment (fchollet) [inline tensorflow/python/keras/backend.py:1537]: You can just in-line this at the usage location

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! This would require some unit tests (especially for the use case that it is fixing). In addition, it would be good to test whether this interferes with mixed precision.

**Comment by bhack** [comment]
Was it reverted?

**Comment by mihaimaruseac** [comment]
No, it wasn't reverted. Just needs tests added in a separate PR.
